### PR TITLE
#10 - Improved the action() function and write tests

### PR DESCRIPTION
**Pull request by kmanaseryan**
Improved the action() function

- [x] Added tests
- [x] Force to sue **GET** if the optional param is `true`
- [x] In case of **GET** request convert the payload to HTTP param and make props snake case

### CODE REVIEW

**Review by mariorodeghiero — CHANGES_REQUESTED** [review]
Please try to use this one

```
async action(actionName, payload, requestType = "POST") {
    const path = `/api/3/action/${actionName}`
    const headers = {
      headers: {
        'Content-Type': 'application/json;charset=utf-8',
        Authorization: this.authToken,
      },
    }

    const response = await axios({
      method: requestType,
      url:`${this.api}${path}`,
      headers: headers,
      data: payload
    })

    if (!response.data.success) {
      throw `Action response: ${response.statusText}`
    }

    return response.data
  }
```

File: lib/index.js [hunk 1]
```diff
@@ -5,17 +5,18 @@ const CkanAuthApi = require('./util/ckan-auth-api')
...
- * @param {string} authToken
+ * @param {string} apiKey
  * @param {number} organizationId
  * @param {string} datasetId
  * @param {string} api
  */
 class Client {
-  constructor(authToken, organizationId, datasetId, api) {
-    this.authToken = authToken
+  constructor(apiKey, organizationId, datasetId, api) {
```
Comment (rufuspollock) [inline lib/index.js:18]: great we have this now - though to be clear my previous comment was an aside and if we change we should probably do in a separate commit to avoid breaking anything and clean PRs.

**Review by rufuspollock — APPROVED** [review]

**Comment by mariorodeghiero** [comment]
@kmanaseryan don't worry about my comment above. I will send this I'm my branch

File: lib/index.js (hunk 1 above)
Comment (kmanaseryan) [reply lib/index.js:18]: 👍
